RevenueCatUI/View+PurchaseRestoreCompleted.swift

```diff
@@ -332,6 +385,50 @@ private struct OnPurchaseCancelledModifier: ViewModifier {
 
 }
 
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct HandlePurchaseAndRestoreModifier: ViewModifier {
+    let performPurchase: PerformPurchase
+    let performRestore: PerformRestore
+
+    func body(content: Content) -> some View {
+        content
+            .modifier(HandlePurchaseModifier(handler: performPurchase))
+            .modifier(HandleRestoreModifier(handler: performRestore))
+    }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+private struct HandlePurchaseModifier: ViewModifier {
+
+    let handler: PerformPurchase
+
+    func body(content: Content) -> some View {
+        content
+            .onPreferenceChange(HandlePurchasePreferenceKey.self) { purchaseResultReporter in
+                if let purchaseResultReporter {
+                    self.handler(purchaseResultReporter.storeProduct, purchaseResultReporter)
+                }
+            }
+    }
+
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+private struct HandleRestoreModifier: ViewModifier {
+
+    let handler: PerformRestore
+
+    func body(content: Content) -> some View {
+        content
+            .onPreferenceChange(HandleRestorePreferenceKey.self) { restoreResultReporter in
+                if let restoreResultReporter {
+                    self.handler(restoreResultReporter)
```

Comment:
The restore method, which is called here, does not provide any information, just a callback which the user can report if it succeeded or not, plus an error. An implementation looks like this:
```swift
}, performRestore: { restoreResultReporter in
    // restore purchases

    // report result to RevenueCat
    restoreResultReporter.reportResult(success: true, error: nil)
})
```
I'm unsure if this is correct, and if we are able to or need to create a correct "CustomerInfo" object after this?

Reply:
I _think_ this is fine! But @aboedo would know for sure 😊

Reply:
I'm a bit lost on the context, but ideally `restore` gives you the latest customer info when it's done

Reply:
Here it's the app developer performing the restore operation (because `purchasesAreCompletedBy == .myApp`), but all they give us back is if the restore succeeded or not (and any associated error). Is that sufficient?

Reply:
It's the code block in my first comment that is the focus here (but this does appear in the PR because that would be customer code, so I commented on the code that triggers it instead, which is probably confusing!).

Reply:
ohh I see, now I understand the initial question, then. Yes, we'd be able to create a CustomerInfo - assuming the SDK is configured (which I'm guessing it is, otherwise it would crash 😅). We would have to internally call `syncPurchases` however if we want CustomerInfo to reflect stuff that the user might have done

Reply:
Ah OK then I will need to add that! And we should do the same after a purchase then too?

Reply:
It should be only for restores - a purchase should still trigger activity in the SK1 queue, which we'll observe, and in SK2 we check purchases after the app comes back into foreground, which should also work.

Reply:
restores are a bit of a special case because nothing actually happens in the queue - when you restore using RC, we just fetch the receipt and send it to the backend

Reply:
restores are a bit of a special case because nothing actually happens in the queue - when you restore using RC, we just fetch the receipt and send it to the backend

Reply:
Perfect, thank you!! 🙏